### DISCUSSION TITLE
Provide a way to synchronize Beets libraries on multiple devices?

### DISCUSSION THREAD

**Discussion by tobast** [beetbox/beets#4464] [Ideas]
Hello,

I've recently started using Beets on a reasonably big library (~130Go, mostly flac). I also routinely use multiple computers (home, work, …) on which I'd like to have all of my music available.

I've searched for a way to keep Beets libraries synchronized on multiple devices, but I came to the conclusion that there is no obvious such way. Most people seem to have their library on one computer, and copy only their audio files over to their other devices.

I'd like to really be able to use beets normally on any of my computers, though. I could probably come up with an ad-hoc solution of `rsync`ing the files and the database over, but it would assume that the database is read-only on all computers but one.

Is there a way that I missed to "merge" databases, possibly outputting a list of files to be synced in both directions? If not, would this look to you like an interesting plugin? From what I can think of right now, it would need to:

* be able to connect to a remote computer (eg. using ssh) — or possibly multiple remote computers at once? — and look up its remote beets library, diff it against the local library, and merge those libraries
* copy over, in both directions, missing files, to keep the libraries synced
* handle removals correctly; ie. if I have computers A and B, delete some album "foo" on `A`, then `beet sync A` on `B`, I don't want this to be interpreted as "aha, A is missing the album foo, let's copy it"
* possibly handle renaming correctly; ie if I have computers A and B, retag and move an album on A, if possible I'd like to avoid deleting + copying over the files upon synchronization, where moving + retagging would be enough

**Comment by arsaboo** [comment]
There is no native solution. Here's what I do. 

My music files are on my Synology NAS and I mount the music files to the same folder on every device `/data/music/`. Once you have the same mount point, you can access it from any number of devices. If you make changes on one of the devices, be sure to run `beet update` on all other devices to sync the updates. 

The other solution (and I have not tried it) may be to save the music library in the same NAS folder and use that library across devices. You cannot run beets simultaneously across devices, but this should work as well.
